bpaotu/bpaotu/views.py

```diff
@@ -495,6 +495,226 @@ def str_none_blank(v):
         return csv_fd.getvalue()
 
 
+
+
+@require_http_methods(["GET"])
+def otu_biom_export(request):
+    SAMPLE_RUN_VAL = 1
+    TRIAL_SWITCH = False
+
+
+
+    def val_or_empty(obj):
+        if obj is None:
+            return ''
+        return obj.value
+
+    def write_output(val, stream, file_ptr):
+        stream.write(val)
+        file_ptr.write(stream.getvalue())
+        stream.seek(0)
+        stream.truncate(0)
+
```

Comment:
remove this function, replace calls to this with a call to `yield`

break out the output code into a new function which just `yield`s strings.

then use the zipfile iterator support (see the other export function) and return a stream of zipped JSON-encoded BIOM data

Reply:
Fixed.